extensions-core/kinesis-indexing-service/src/test/java/org/apache/druid/indexing/kinesis/supervisor/KinesisSupervisorTest.java

```diff
@@ -3824,6 +4035,100 @@ public void testIsTaskCurrent()
     verifyAll();
   }
 
+  @Test
+  public void testTimeLagMetricsDisabled() throws Exception
```

Comment:
With regards to disabling the time lag metric, what behavior is tested by `testDiscoverExistingPublishingAndReadingTaskWithoutTimeLagMetrics` that is not covered by this test?

What do you think about refactoring some of the logic here so that it's reused by `testNoInitialState`?

Reply:
`testDiscoverExistingPublishingAndReadingTaskWithoutTimeLagMetrics` was checking for time lag metrics in the active tasks portion of the supervisor status report, which `testNoInitialState` didn't do.

`testNoInitialState` had checks for time lag metrics in the aggregate-level portion of the supervisor report which `testDiscoverExistingPublishingAndReadingTaskWithoutTimeLagMetrics` didn't

I ended up deleting `testTimeLagMetricsDisabled` and collapsed the time lag metrics tests into `testDiscoverExistingPublishingAndReadingTask*` test pair which now checks all 3 areas.